### PR TITLE
Add firebase sign up with email and password

### PR DESCRIPTION
**Pull request by carloseag1609**
# Description
 - We've created the file firebaseSignIn.js in a new helpers folder. It exports a function which creates a new user on firebase.
 - We've fixed the two checkbox components in SignUp screen. Why? The checkboxes didn't work correctly (never checks or unchecks) and one of them didn't correspond with the form validations, so we fixed it.
 - We've fixd  the Yup validation schema with a valid regular expression for the password. Why? The regular expression for the password validation in the schema was a string, so we changed it to a valid regular expression.
 - We've added a new "loading" state. Why? When the form is submitted and called the function that creates the new user, it takes some time, so we added a new state (loading) which indicates if the user creation is loading or finished.

### CODE REVIEW

**Review by cortezroberto — COMMENTED** [review]
It's ok.

**Review by yovanha — DISMISSED** [review]
Great!

File: FlightReservation/src/screens/SignUp.js [hunk 1]
```diff
@@ -1,118 +1,142 @@
...
+              </View>
+              <View>
+                <Text>Email</Text>
+                <TextInput
+                  onChangeText={handleChange('email')}
+                  onBlur={handleBlur('email')}
+                  value={values.email}
+                />
+                {errors.email && touched.email && <Text>{errors.email}</Text>}
+              </View>
```
Comment (DemisRincon) [inline FlightReservation/src/screens/SignUp.js:85]: Most of the inputs, title, and error text could be extracted as a component, tools of code improvement like CodeClimate will detect this as code repeated.
The rest of the code looks great.

**Review by jkzasori — CHANGES_REQUESTED** [review]
It's okay but... I think you could refactor FirebaseSignIn component because has a lot of code and with the time it will not be easy to support components with a lot of code also I think you can reuse code.

**Review by DemisRincon — CHANGES_REQUESTED** [review]
Please review the recommendation, in general good!

**Comment by carloseag1609** [comment]
Thanks for the comments :) We'll refactor these inputs and texts in the next feature with the styles.

**Review by DemisRincon — APPROVED** [review]

**Review by DemisRincon — APPROVED** [review]

**Review by DemisRincon — APPROVED** [review]

**Review by cortezroberto — APPROVED** [review]
it's ok

**Review by yovanha — APPROVED** [review]
Good changes!

**Pull request merged**